code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/PushTemplateImageUtil.kt

```diff
@@ -42,105 +43,158 @@ internal object PushTemplateImageUtil {
     private const val FULL_BITMAP_QUALITY = 100
     private const val DOWNLOAD_TIMEOUT_SECS = 10
 
-    private val executor: ExecutorService by lazy { Executors.newSingleThreadExecutor() }
-
-    private class DownloadImageCallable(val url: String?) :
-        Callable<Bitmap?> {
-        override fun call(): Bitmap? {
-            var bitmap: Bitmap? = null
-            var connection: HttpURLConnection? = null
-            try {
-                val imageUrl = URL(url)
-                connection = imageUrl.openConnection() as HttpURLConnection
-                connection.inputStream.use { inputStream ->
-                    bitmap = BitmapFactory.decodeStream(inputStream)
+    /**
+     * Downloads an image using the provided url `String`. Prior to downloading, the image url
```

Comment:
Recommend specifying that  this is a blocking method that returns only after all downloads have finished either by failing or successfully being downloaded.